flake8_type_checking/checker.py

```diff
@@ -492,6 +614,14 @@ def names(self) -> set[str]:
         """Return unique names."""
         return set(self.uses.keys())
 
+    def type_checking_symbols(self) -> Iterator[Symbol]:
+        """Yield all the symbols declared inside a type checking block."""
+        for scope in self.scopes:
+            for symbols in scope.symbols.values():
+                for symbol in symbols:
```

Comment:
Depends on what you mean by unique. The same name can occur multiple times, since a name can be redefined, but the symbol object itself is unique, yes and it includes its location so we can reason about where it is available and where it isn't available.